feature/text_based.py

```diff
@@ -0,0 +1,703 @@
+        if matrix is None:
+            raise ValueError("Matrix cannot be None.")
+        if matrix.ndim != 2:
+            raise ValueError("Matrix should be 2D")
+        if len(cost) != matrix.shape[1]:
+            raise ValueError("Cost vector size should match the number of columns in the matrix: "
+                             f"{len(cost)} vs. {matrix.shape[1]}.")
+
+
+class ContentSelector:
+    """ Content Selector.
+
+    This module contains classes for feature selection, in which
+    a subset of features is selected from a larger set of features
+    in order to maximize the performance of a given model or analysis.
+
+    The module includes four selection algorithms:
+        Random selection (random),
+        Greedy forward selection (greedy),
+        KMeans clustering-based selection (KMeans), and
+        Multi-level set-covering optimization-based approach (exact).
+
+    All algorithms can be used with either a uniform cost metric (unicost) or
+    a diversity-based cost metric (diversity).
+
+    The general optimization process involves selecting a subset of features
+    that maximizes diversity in the text embedding spaces and label coverage
+    while minimizing the number of features selected.
+
+    Attributes
+    ----------
+    selection_size: int
+        Number of content/feature to select.
+    seed: int
+        Random seed.
+    trials: int
+        Number of trials to run for random functions to estimate average coverage.
+    verbose: bool
+        Print intermediate results if True.
+    matrix: np.ndarray
+        Coverage matrix where each row corresponds to all predefined labels for each category and the columns
+        correspond to the content.
+    features: np.ndarray
+        Numeric featurization of content.
+
+    References:
+    [1] Kadioglu et al., Optimized Item Selection to Boost Exploration for Recommender Systems, CPAIOR'21
+    [2] Kleynhans et al. Active Learning Meets Optimized Item Selection, DSO@IJCAI'21
+
+    """
+
+    def __init__(self, selection_size: int, seed: int = Constants.default_seed, trials: int = 10,
+                 verbose: bool = False):
+        """ContentSelector
+
+       Creates a content selector object.
+
+       Parameters
+       ----------
+        selection_size: int
+            Maximum number of content to select.
+        seed: int, default=123456
+            Random seed.
+        trials: int, default=10
+            Number of trials to run for random functions to estimate average coverage.
+        verbose: bool, default=False
+            Print intermediate results if True.
+        """
+
+        # Validate arguments
+        self.featurization_method = None
+        self.cost_metric = None
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = False
```

Comment:
shoudl not this be set to verbose? from the constructor

Reply:
solved!